pkg/cmd/cli/cmd/set/triggers.go

```diff
@@ -0,0 +1,744 @@
+package set
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/generate/app"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+const (
+	triggersLong = `
+Set or remove triggers for build configs and deployment configs
+
+All build configs and deployment configs may have a set of triggers that result in a new deployment
+or build being created. This command enables you to alter those triggers - making them automatic or
+manual, adding new entries, or changing existing entries.
```

Comment:
I think the normal operation is --manual and --auto, and --remove is an
unusual case.  I want to toggle, not destroy data.

On Tue, Feb 23, 2016 at 11:18 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/set/triggers.go
> <https://github.com/openshift/origin/pull/7423#discussion_r53892268>:
>
> > +	buildutil "github.com/openshift/origin/pkg/build/util"
> > +	cmdutil "github.com/openshift/origin/pkg/cmd/util"
> > +	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
> > +	deployapi "github.com/openshift/origin/pkg/deploy/api"
> > +	"github.com/openshift/origin/pkg/generate/app"
> > +	imageapi "github.com/openshift/origin/pkg/image/api"
> > +	"k8s.io/kubernetes/pkg/util/sets"
> > +)
> > +
> > +const (
> > +	triggersLong = `
> > +Set or remove triggers for build configs and deployment configs
> > +
> > +All build configs and deployment configs may have a set of triggers that result in a new deployment
> > +or build being created. This command enables you to alter those triggers - making them automatic or
> > +manual, adding new entries, or changing existing entries.
>
> so setting a buildconfig trigger to "manual" deletes the trigger, unless
> it's a webhook trigger in which case it does nothing?
>
> I feel like it would be better if --manual was just an error for
> buildconfigs and the user got told to invoke it with --remove instead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7423/files#r53892268>.
>